### PR TITLE
Wemo binary switch translator

### PR DESCRIPTION
**Pull request by danielrischar**
Adds a wemo binary switch translator.

The wemo switch is a proximal WiFi device

### CODE REVIEW

File: org.opent2t.sample.binaryswitch.superpopular/com.wemo.binaryswitch/js/thingTranslator.js [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+    } else if (typeof arg !== expectedType) {
+        throw new Error('Invalid argument: ' + argName + '. ' +
+            'Expected type: ' + expectedType + ', got: ' + (typeof arg));
+    }
+}
+
+/**
+ * Helper method to convert the provider schema to the platform schema.
+ */ 
+function providerSchemaToPlatformSchema(providerState, deviceInfo, expand) {
```
Comment (shalinijoshi19) [inline org.opent2t.sample.binaryswitch.superpopular/com.wemo.binaryswitch/js/thingTranslator.js]: Nit: Helpers i thought we were following the convention of pulling these at the bottom of the file + naming them starting wiht an '_'

File: org.opent2t.sample.binaryswitch.superpopular/com.wemo.binaryswitch/js/thingTranslator.js [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+    });
+
+    var resource = entity.resources.find((r) => {
+        return r.id === resourceId;
+    });
+
+    return resource;
+}
+
+// Each device in the platform has its own static identifier
```
Comment (shalinijoshi19) [inline org.opent2t.sample.binaryswitch.superpopular/com.wemo.binaryswitch/js/thingTranslator.js]: >tatic identif [](start = 44, length = 13)

Who uses this deviceId today up the chain?

File: org.opent2t.sample.binaryswitch.superpopular/com.wemo.binaryswitch/js/thingTranslator.js [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+     * Post the power resource to the binary switch
+     */
+    postDevicesPower(di, payload) {
+        return this._postDeviceResource(di, 'power', payload);
+    }
+
+    /** 
+     * Subscribe to state changes to the binary switch 
+     * */
+    postSubscribe(subscriptionInfo) {
```
Comment (shalinijoshi19) [inline org.opent2t.sample.binaryswitch.superpopular/com.wemo.binaryswitch/js/thingTranslator.js]: is this ever going to be supported?

**Comment by shalinijoshi19** [comment]
:shipit:

**Comment by arjun-msft** [comment]
:shipit:

**Comment by danielrischar** [comment]
Tabling this for now.

**Pull request closed without merging**